### ISSUE TITLE
@Schemable reports a false initializer mismatch when a compatible overload appears later

### ISSUE THREAD

**Issue by ajevans99** [ajevans99/swift-json-schema#191]
## Summary

`@Schemable` emits a compiler error for a valid model when an incompatible initializer overload appears before a compatible overload with the same parameter names/count. Reordering the declarations makes the error disappear.

## Reproduction

```swift
import JSONSchemaBuilder

@Schemable
struct Token {
  let value: Int

  init(value: String) {
    self.value = Int(value) ?? 0
  }

  init(value: Int) {
    self.value = value
  }
}
```

The macro emits:

> Parameter 'value' has type 'String' but schema expects 'Int'. This type mismatch will cause the generated schema to fail.

This diagnostic has `.error` severity. Placing `init(value: Int)` first avoids the false diagnostic.

Draft #189 contains [planner and macro-expansion tests for both declaration orders](https://github.com/ajevans99/swift-json-schema/blob/da28cfb24f8236f8f312672f9465a6433d0b3f06/Tests/JSONSchemaMacroTests/SchemaPlanningTests.swift#L283). The erroneous case is supplied as source text to the test harness, so the test target itself still compiles.

```sh
swift test --build-system native --force-resolved-versions --jobs 6 \
  --filter 'SchemaPlanningTests/initializerOverloadMatchingIgnoresDeclarationOrder'
```

## Expected behavior

Both declaration orders should generate the integer-valued schema without a false initializer diagnostic. A compatible `init(value: Int)` exists and can be selected for the generated `JSONSchema(Token.init)` expression.

## Actual behavior

- String-first order fails both the planner's no-diagnostics assertion and the expansion harness's expected-empty diagnostic list.
- Int-first order passes.
- The compatible-first compiled runtime integration also parses integer input correctly.

These are two observations of one selection defect, not two independent bugs. The finding is a false compile-time diagnostic, not proof that generated code silently invokes the wrong overload at runtime.

## Cause

After #182, the implementation lives in `Sources/JSONSchemaMacro/Schemable/Diagnostics/InitializerDiagnostics.swift`:

- `findMatchingInit` returns the first initializer whose parameter count and parameter-name set match, without comparing types.
- `validateInitParameters` then checks only that selected initializer and emits the mismatch.
- The new normalized type comparison recognizes equivalent spellings, but it does not change candidate selection.

## Acceptance criteria

- Consider compatible types and parameter order before selecting an initializer to validate.
- Both declaration-order regressions pass without suppressing diagnostics.
- Preserve real mismatch diagnostics when no compatible initializer exists and retain #182's normalized type-spelling behavior.

## Context

Reproduced after merged #182 (`ecdf222`), on Apple Swift 6.4/macOS. Tests-only draft #189 contains the regression coverage; no production fix has been made.
